### ISSUE TITLE
Why keep only the first mask in the batch?

### ISSUE THREAD

**Issue by CharlesNord** [daa233/generative-inpainting-pytorch#31]
https://github.com/daa233/generative-inpainting-pytorch/blob/ee1fd75f5253b579cd12ede8b7ed27f3ee6e747d/model/networks.py#L262

Hello, sorry to bother you. Could you please explain why you keep only the first mask in the batch, in your implementation of the contextual attention module?

**Comment by CharlesNord** [comment]
I wonder if the follow block:
`
        m = m[0]    # m shape: [L, C, k, k]

        # mm shape: [L, 1, 1, 1]

        mm = (reduce_mean(m, axis=[1, 2, 3], keepdim=True)==0.).to(torch.float32)

        mm = mm.permute(1, 0, 2, 3) # mm shape: [1, L, 1, 1]`

 should be moved into the for loop:
`for xi, wi, raw_wi in zip(f_groups, w_groups, raw_w_groups):`

**Comment by daa233** [comment]
It follows the original TensorFlow implementation [here](https://github.com/JiahuiYu/generative_inpainting/blob/v1.0.0/inpaint_ops.py#L270). Here is a discussion about it: https://github.com/JiahuiYu/generative_inpainting/issues/85

Actually, I have tried both you mentioned above. It seems that using different masks in a batch converges a little slower but I am not very sure about it.

**Comment by CharlesNord** [comment]
> It follows the original TensorFlow implementation [here](https://github.com/JiahuiYu/generative_inpainting/blob/v1.0.0/inpaint_ops.py#L270). Here is a discussion about it: [JiahuiYu/generative_inpainting#85](https://github.com/JiahuiYu/generative_inpainting/issues/85)
> 
> Actually, I have tried both you mentioned above. It seems that using different masks in a batch converges a little slower but I am not very sure about it.

Thank you for your explanation, now I understand, that's because the masks in one batch are the same.
